include/libcyphal/transport/can/delegate.hpp

```diff
@@ -0,0 +1,218 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_TRANSPORT_CAN_DELEGATE_HPP_INCLUDED
+#define LIBCYPHAL_TRANSPORT_CAN_DELEGATE_HPP_INCLUDED
+
+#include <libcyphal/transport/errors.hpp>
+#include <libcyphal/transport/scattered_buffer.hpp>
+
+#include <cetl/rtti.hpp>
+
+#include <canard.h>
+
+namespace libcyphal
+{
+namespace transport
+{
+namespace can
+{
+namespace detail
+{
+
+class TransportDelegate
+{
+    // 1141F5C0-2E61-44BF-9F0E-FA1C518CD517
+    using CanardMemoryTypeIdType = cetl::
+        type_id_type<0x11, 0x41, 0xF5, 0xC0, 0x2E, 0x61, 0x44, 0xBF, 0x9F, 0x0E, 0xFA, 0x1C, 0x51, 0x8C, 0xD5, 0x17>;
+
+public:
+    class CanardMemory final : public cetl::rtti_helper<CanardMemoryTypeIdType, ScatteredBuffer::Interface>
```

Comment:
Maybe `CanardScatteredBuffer` makes a better name? The current name is a bit too generic.

Reply:
maybe... in the next pr I will already have `ScatteredBuffer` (renamed `DynamicBuffer`)